### PR TITLE
Adds "SaveAs" menu item in PluginCreator

### PR DESCRIPTION
**Pull request by rjmatthews62**
New puts in correct version of MuseScore for Import.

### CODE REVIEW

File: mscore/pluginCreator.cpp [hunk 1]
```diff
@@ -430,15 +436,16 @@ void PluginCreator::load()
 //   savePlugin
 //---------------------------------------------------------
 
-void PluginCreator::savePlugin()
-      {
-      if (created) {
+void PluginCreator::doSavePlugin(bool saveas) 
+{
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: Differs from the preferred banner style indent. Here and several times below too

File: mscore/pluginCreator.h [hunk 2]
```diff
@@ -43,20 +43,21 @@ class PluginCreator : public QMainWindow, public Ui::PluginCreatorBase {
...
       void savePlugin();
+      void savePluginAs();
       void newPlugin();
       void textChanged();
       void closePlugin();
       void showManual();
       void qmlWarnings(const QList<QQmlError>&);
-
-   signals:
+signals:
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.h]: please keep indent as it was here

File: mscore/revision.h [hunk 3]
```diff
@@ -1 +1 @@
-3543170
+c1a89a0
```
Comment (Jojo-Schmitz) [inline mscore/revision.h]: Don't commit changes to this file, ever.
In your clone directory, copy the file build/git/hooks/post-checkout to the directory .git/hooks in order for mscore/revision.h to be maintained automatically by git. See build/git/hooks/README for details. Note that the .git directory is hidden.

**Comment by Jojo-Schmitz** [comment]
Did you sign the [CLA](https://musescore.org/cla), if not, please do so.

**Comment by rjmatthews62** [comment]
CLA has been signed.
Did you want me to corrected indenting etc, or worry about it next time?

**Comment by Jojo-Schmitz** [comment]
Depends on what you mean by 'next time', it certainly needs to be fixed before this PR gets accepted and merged

**Comment by rjmatthews62** [comment]
That kinda was the question. Ok, I'll make the corrections and try again.

File: mscore/pluginCreator.cpp [hunk 4]
```diff
@@ -480,7 +497,7 @@ void PluginCreator::newPlugin()
       created = true;
       QString s(
          "import QtQuick 2.0\n"
-         "import MuseScore 1.0\n"
+         "import MuseScore 3.0\n"
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: This part might better be in a separate commit, as this applies to master only, the other stuff might apply to 2.2 too.
But I'd rather have @lasconic voice his opinion here ;-)

Comment (lasconic) [reply mscore/pluginCreator.cpp]: Thanks for the heads up, I can fix that when cherry picking, no pb. So no need for 2 commits

**Comment by rjmatthews62** [comment]
I'm trying to work out how to do that (revision.h, single commit...) 

Also trying to work out where you think I've gone wrong with the indenting... thought I'd checked everything.

**Comment by Jojo-Schmitz** [comment]
One way to fix the commits:
`git reset <SHA before your first commit>`
`git checkout -- revision.h`
`git commit -m"<message>"`
`git push -f`

**Comment by rjmatthews62** [comment]
Ta. I've also worked out the issue with the tabs... had to be firm with my editor, it was sticking them in without permission.

**Comment by Jojo-Schmitz** [comment]
To get the coding style right check the qt2-musescore.xml file at https://musescore.org/en/developers-handbook-finding-your-way-around/musescore-coding-rules#Coding_style

File: mscore/pluginCreator.cpp [hunk 5]
```diff
@@ -467,6 +467,16 @@ void PluginCreator::doSavePlugin(bool saveas)
       }
 
 void PluginCreator::savePlugin()
+{
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: still wrong brace indent...

File: mscore/pluginCreator.cpp [hunk 6]
```diff
@@ -430,15 +436,16 @@ void PluginCreator::load()
...
-void PluginCreator::savePlugin()
+void PluginCreator::doSavePlugin(bool saveas) 
       {
-      if (created) {
+      if (saveas) {
             path = mscore->getPluginFilename(false);
             if (path.isEmpty())
                   return;
-            }
+      }
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: indent...

File: mscore/pluginCreator.cpp [hunk 7]
```diff
@@ -452,13 +459,23 @@ void PluginCreator::savePlugin()
             setState(PCState::CLEAN);
             setTitle( fi.completeBaseName() );
             setToolTip(path);
-            }
+      }
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: indent...

File: mscore/pluginCreator.cpp [hunk 8]
```diff
@@ -452,13 +459,23 @@ void PluginCreator::savePlugin()
...
       else {
             // TODO
-            }
+      }
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.cpp]: indent...

File: mscore/pluginCreator.h [hunk 9]
```diff
@@ -43,19 +43,20 @@ class PluginCreator : public QMainWindow, public Ui::PluginCreatorBase {
...
       void loadPlugin();
       void load();
       void savePlugin();
+      void savePluginAs();
       void newPlugin();
       void textChanged();
       void closePlugin();
       void showManual();
       void qmlWarnings(const QList<QQmlError>&);
-
```
Comment (Jojo-Schmitz) [inline mscore/pluginCreator.h]: better leave that blank line in

**Comment by rjmatthews62** [comment]
This Banner style seems unnatural and wrong... :P

File: mscore/pluginCreator.cpp (hunk 6 above)
Comment (Jojo-Schmitz) [reply mscore/pluginCreator.cpp]: you forgot tzhis one (but rightfully fixed another and older one a few lines down)

**Comment by Jojo-Schmitz** [comment]
No worries, we'll get there, these kind of teething problems happen to almost everybody on their first PR ;-)

**Comment by rjmatthews62** [comment]
This is literally the first time I've ever done a PR... all my prior uses of git were inhouse... I didn't even realize the push would automatically update.

**Comment by lasconic** [comment]
Looks good except for the line highlighted by Jojo.
When done, you can squash the commits together and force push.
See http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by rjmatthews62** [comment]
I think that's right? Did my best to squash it down, not sure I succeeded.

**Comment by Jojo-Schmitz** [comment]
As far as I can see it is still 2 commits and revision.h is still part of it

**Comment by rjmatthews62** [comment]
Cant work out how to tell git to drop revision.h. 
May just cancel this and try again.

**Comment by Jojo-Schmitz** [comment]
As mentioned further up:

`git reset <SHA before your first commit>`
`git checkout -- revision.h`
`git commit -m"<message>"`
`git push -f`

**Comment by rjmatthews62** [comment]
I so wish I knew what I did there...
